addons/crm_addons/controllers/main.py

```diff
@@ -0,0 +1,566 @@
+                    'name': l.name
+                })
+
+        return {
+            'authorized': True,
+            'success': not len(errors),
+            'errors': errors
+        }
+
+    def _get_leads_for_partner_id(self, request, id, offset, limit):
+        partner_leads = request.env['crm.lead'].with_user(request.env.user).search([('partner_id', '=', id)], offset=offset, limit=limit)
+        leads = []
+        for lead in partner_leads:
+            leads.append({
+                'id': lead.id,
+                'name': lead.name,
+                'expected_revenue': str(lead.expected_revenue),
+                'planned_revenue': str(lead.planned_revenue),
+                'currency_symbol': lead.company_currency.symbol
+            })
+        return leads
+
+    @http.route('/mail_client_extension/lead/get_by_partner_id', type="json", auth="none", csrf=False, cors="*")
+    def crm_lead_get_by_partner_id(self, **kwargs):
+        if not CrmAddonsController._authorize(request):
+            return {'authorized': False}
+
+        partner_id = request.jsonrequest.get("partner_id")
+        limit = request.jsonrequest.get("limit")
+        offset = request.jsonrequest.get("offset")
+        return {
+            'authorized': True,
+            'leads': self._get_leads_for_partner_id(request, partner_id, offset=offset, limit=limit)
+        }
+
+
+    @http.route('/mail_client_extension/lead/create', type="json", auth="none", csrf=False, cors="*")
+    def crm_lead_create(self, **kwargs):
+        if not CrmAddonsController._authorize(request):
+            return {'authorized': False}
+
+        lead_values = request.jsonrequest#.get("lead_values")
+        lead = request.env['crm.lead'].with_user(request.env.user).create({
+            'name': lead_values['name'],
+            'partner_id': int(lead_values['partner_id']),
+            'planned_revenue': int(lead_values['expected_revenue']),
+            'priority': lead_values['priority']
+        })
+
+        return {
+            'authorized': True,
+            'success': True if lead else False
+        }
+
+    #@http.route('/crm/addons/lead/delete', type="json", auth="none", csrf=False, cors="*")
+    def _crm_lead_delete(self, **kwargs):
+        if not CrmAddonsController._authorize(request):
+            return {'authorized': False}
+
+        lead_id = request.jsonrequest.get("id")
+        lead = request.env['crm.lead'].with_user(request.env.user).search([('id', '=', lead_id)])
+        if not lead:
+            return {
+                'success': False
+            }
+
+        partner_id = lead.partner_id.id
+        lead.unlink()
+
+        return {
+            'authorized': True,
+            'success': True,
+            'leads': self._get_leads_for_partner_id(request, partner_id)
+        }
+
+    @http.route('/mail_client_extension/lead/redirect_form_view', type='http', auth='user', methods=['GET'])
+    def crm_lead_redirect_form_view(self, partner_id):
```

Comment:
The naming of the endpoint and method seem less than good, they should probably refer to the partne somehow e.g. `lead/create_from_partner` or somesuch?